src/cli/net/client/mod.rs

```diff
@@ -38,25 +48,63 @@ impl MpcHandle for MpcHttpConnection {
 #[allow(dead_code)]
 impl MpcHttpConnection {
     #[must_use]
-    pub fn new(addr: &str) -> Self {
+    pub fn new(addr: Uri) -> Self {
         // this works for both http and https
         let https = HttpsConnector::new();
         let client = Client::builder().build::<_, Body>(https);
 
         Self {
             client,
-            addr: addr.parse().expect("Cannot parse the URI"),
+            addr_parts: addr.into_parts(),
```

Comment:
I'd rather keep scheme and authority in the state: `scheme: Scheme, authority: Authority`. you wouldn't need to unwrap these values each time you want to access them

Reply:
done